From 2ccfe2c5373ea851137d2152bc9aabb7dda1f132 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADt=20Ondruch?= <vondruch@redhat.com>
Date: Thu, 7 Jan 2021 12:41:31 +0100
Subject: [PATCH] Drop safe level support in more recent Rubies.

The `rb_set_safe_level` is available approximately since Ruby 1.8.0, but the
RUBY_SAFE_LEVEL_MAX was available since Ruby 2.1, until it was dropped
in Ruby 3.0 altogether with Ruby safe levels [[1]].

[1]: https://github.com/ruby/ruby/commit/e91c39f1c0f7d5e670266d9593d533fd444957f6
---
 ruby/rubyinterpreter.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ruby/rubyinterpreter.cpp b/ruby/rubyinterpreter.cpp
index 3bccdc2..ad236d5 100644
--- a/ruby/rubyinterpreter.cpp
+++ b/ruby/rubyinterpreter.cpp
@@ -65,8 +65,11 @@ RubyInterpreter::RubyInterpreter(Kross::InterpreterInfo* info)
         initRuby();
     }
 
+// The RUBY_SAFE_LEVEL_MAX with the safe levels were removed in Ruby 3.0.
+#ifdef RUBY_SAFE_LEVEL_MAX
     const int defaultsafelevel = 4; // per default use the maximum safelevel
     rb_set_safe_level( info->optionValue("safelevel", defaultsafelevel).toInt() );
+#endif
 }
 
 RubyInterpreter::~RubyInterpreter()
-- 
2.29.2

